addons/xivbar/events.lua

```diff
@@ -0,0 +1,92 @@
+--[[
+        Copyright © 2017, SirEdeonX
+        All rights reserved.
+
+        Redistribution and use in source and binary forms, with or without
+        modification, are permitted provided that the following conditions are met:
+
+            * Redistributions of source code must retain the above copyright
+              notice, this list of conditions and the following disclaimer.
+            * Redistributions in binary form must reproduce the above copyright
+              notice, this list of conditions and the following disclaimer in the
+              documentation and/or other materials provided with the distribution.
+            * Neither the name of xivbar nor the
+              names of its contributors may be used to endorse or promote products
+              derived from this software without specific prior written permission.
+
+        THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+        ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+        WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+        DISCLAIMED. IN NO EVENT SHALL SirEdeonX BE LIABLE FOR ANY
+        DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+        (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+        LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+        ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+        (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+        SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+-- Bind Events
+-- ON LOAD
+windower.register_event('load',function()
+    if windower.ffxi.get_info().logged_in then
+        initialize()
+    end
+end)
+
+-- ON LOGIN
+windower.register_event('login',function()
+    initialize()
+end)
+
+-- ON LOGOUT
+windower.register_event('logout',function()
+    hide()
+end)
+
+-- BIND EVENTS
+windower.register_event('hp change', function(new, new)
+    hp_update = true
+end)
+
+windower.register_event('hpmax change', function(new, old)
+    hp_update = true
+end)
+
+windower.register_event('mp change', function(new, old)
+    mp_update = true
+end)
+
+windower.register_event('mpmax change', function(new, old)
+    mp_update = true
+end)
+
+windower.register_event('tp change', function(new, old)
+    tp_update = true
+end)
+
+windower.register_event('prerender', function()
```

Comment:
I timed a few of the functions a while ago. I think I looped the functions 5,000 times.

get_items (no args):   6.377
get_player:                0.677
get_items(0):             0.483
get_party (6 players): 0.322
get_party (1 players): 0.167
get_mob_by_id:         0.035 (full zone)
get_mob_by_id:         0.030 (empty zone)
get_mob_by_index:    0.028